### PR TITLE
Move `bench-utils` into `std`, and simplify workspace

### PR DESCRIPTION
**Pull request by Pratyush**
This PR merges `bench-utils` into `std`, thus simplifying the dependency story for downstream crates. It also makes it easier to publish things on `crates.io`. cc @ValarDragon @weikengchen

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,29 +1,29 @@
...
-    "std",
-]
+[dependencies]
+rand = { version = "0.7", default-features = false }
+# rand = { version = "0.8", default-features = false, features = ["std_rng"] }
+rand_xorshift = "0.2"
+# rand_xorshift = "0.3"
+rayon = { version = "1", optional = true }
 
-[profile.release]
```
Comment (ValarDragon) [inline Cargo.toml:8]: Don't we still want these profiles?

**Comment by ValarDragon** [comment]
LGTM sans the question about build profiles

File: Cargo.toml (hunk 1 above)
Comment (Pratyush) [reply Cargo.toml:8]: Oops, you're right. Thanks for catching that.

**Comment by Pratyush** [comment]
Ok let's merge this after the relevant PRs are made to the dependant crates.

**Review by ValarDragon — APPROVED** [review]

**Comment by Pratyush** [comment]
btw, should we rename this repo to `std`?

**Comment by weikengchen** [comment]
I agree with the renaming since the main one we use is `ark_std` anyway.

**Review by weikengchen — APPROVED** [review]

**Pull request merged**